src-java/kilda-pce/src/main/java/org/openkilda/pce/AvailableNetworkFactory.java

```diff
@@ -65,18 +65,16 @@ public AvailableNetwork getAvailableNetwork(Flow flow, Collection<PathId> reuseP
             // Reads all active links from the database and creates representation of the network.
             getAvailableIsls(buildStrategy, flow)
                     .forEach(link -> addIslAsEdge(link, network));
-
-            if (!reusePathsResources.isEmpty() && !flow.isIgnoreBandwidth()) {
-                reusePathsResources.stream()
-                        .filter(pathId -> flowPathRepository.findById(pathId)
-                                .map(path -> !path.isIgnoreBandwidth())
-                                .orElse(false))
-                        .forEach(pathId -> {
-                            // ISLs occupied by the flow (take the bandwidth already occupied by the flow into account).
-                            islRepository.findActiveByPathAndBandwidthAndEncapsulationType(
-                                    pathId, flow.getBandwidth(), flow.getEncapsulationType())
-                                    .forEach(link -> addIslAsEdge(link, network));
-                        });
+            if (!flow.isIgnoreBandwidth()) {
+                if (flow.getYFlowId() != null) {
+                    Collection<PathId> flowPaths =
+                            flowPathRepository.findPathIdsBySharedBandwidthGroupId(flow.getYFlowId());
+                    if (!flowPaths.isEmpty()) {
+                        reuseResources(flowPaths, flow, network);
+                    }
```

Comment:
I know that it works, but this code looks like not very correct. If we are working with Y flow we ignore `reusePathsResources` parameter. And this looks incorrect from user point of view. 
If user use `reusePathsResources` collection then he expects that this paths will be ignored.

I think we have several options here:
1. collected shared paths of Y flow in FlowHs topology and call getAvailableNetwork with list of Y flow path list
2. add some other method getAvailableNetwork only for Y flows
3. rewrite getAvailableNetwork to use repositories for finding reusePathsResources, but I think it's not possible, because sometimes we need to force the method to do not use shared resources. For example we use it in reroute operation: 
[reusePathsResources](https://github.com/telstra/open-kilda/blob/37ce148c2e974d3840c3e9a92cb238850ff07a0d/src-java/flowhs-topology/flowhs-storm-
topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/FlowRerouteFsm.java#L113)

Reply:
Done. 
As discussed offline, I've joined both lists if it is YFlow instead of not using the reusePathResources.

Reply:
Yes, much better. Even we can remove the if (!reusePaths.isEmpty()) since the reuseResources iterate over the Set, so if it is empty it doesn't iterate.